### PR TITLE
Enforce "max chunks per query" limit in ingesters too

### PR DESCRIPTION
**Pull request by pracucci**
**What this PR does**:
The `-store.query-chunk-limit` is only enforced in the long-term storage (but not ingesters). However, there are cases where this limit can actually be hit just querying ingesters, so I would like to apply the limit when fetching chunks from ingesters too.

However, if we extend `-store.query-chunk-limit` to apply to ingesters too, that could be considered a breaking change, so in this PR I'm proposing to deprecate `-store.query-chunk-limit` in favour of a new config option `-querier.max-fetched-chunks-per-query`, whose limit is applied both to ingesters and long-term storage.

Due to how the querying works in Cortex (ingesters and long-terms storage are queried separately) in this PR I'm proposing to keep it simple and apply the limit independently to ingesters and storage. This means that the total number of actual fetched chunks could be 2x the limit, being independently applied when querying ingesters and long-term storage. If accepted, this is something we may get back and eventually improve (in my opinion having the limit applied to ingesters too is better than having a true global limit, considering ingesters are currently unbounded, but I'm open to feedback). 

**Which issue(s) this PR fixes**:
N/A

**Checklist**
- [x] Tests updated
- [ ] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: pkg/util/labels_test.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+			input: []*labels.Matcher{
+				labels.MustNewMatcher(labels.MatchEqual, "foo", "bar"),
+			},
+			expected: `{foo="bar"}`,
+		}, {
+			input: []*labels.Matcher{
+				labels.MustNewMatcher(labels.MatchEqual, "foo", "bar"),
+				labels.MustNewMatcher(labels.MatchNotEqual, "who", "boh"),
+			},
+			expected: `{foo="bar",who!="boh"}`,
```
Comment (pstibrany) [inline pkg/util/labels_test.go:28]: Test with `__name__` label name would show that no special handling of that name is done.

**Comment by gouthamve** [comment]
Can you please rebase against master to pull in https://github.com/cortexproject/cortex/pull/4137 and move the changelog entry to the top?

**Review by pstibrany — APPROVED** [review]
Good job, nothing to complain about. :)

**Comment by pracucci** [comment]
Thanks @pstibrany, I've addressed your feedback!

**Pull request merged**